tests/core/test_message.c

```diff
@@ -88,44 +88,48 @@ static byte_t* create_signature_unlock_block() {
-                                         0xfc, 0x68, 0xf5, 0x97, 0xeb, 0x33, 0x93, 0xaf, 0x80, 0xf7, 0x4c,
-                                         0x7c, 0x3d, 0xb7, 0x81, 0x98, 0x14, 0x7d, 0x5f, 0x1f, 0x92};
-
-  ed25519_keypair_t seed_keypair = {};
-  TEST_ASSERT(hex_2_bin("f7868ab6bb55800b77b8b74191ad8285a9bf428ace579d541fda47661803ff44", 64, seed_keypair.pub,
-                        ED_PUBLIC_KEY_BYTES) == 0);
-  TEST_ASSERT(
-      hex_2_bin("256a818b2aac458941f7274985a410e57fb750f3a3a67969ece5bd9ae7eef5b2f7868ab6bb55800b77b8b74191ad8285"
-                "a9bf428ace579d541fda47661803ff44",
-                128, seed_keypair.priv, ED_PRIVATE_KEY_BYTES) == 0);
-
-  core_message_t* msg = core_message_new();
-  TEST_ASSERT_NOT_NULL(msg);
+  byte_t expected_essence_hash[CRYPTO_BLAKE2B_HASH_BYTES] = {
+      0x2,  0x27, 0x34, 0x34, 0x93, 0xb3, 0x9f, 0xdb, 0x33, 0xbb, 0xf9, 0x2c, 0x5a, 0x4a, 0xcd, 0x3c,
+      0xbc, 0x44, 0xae, 0xe2, 0xa2, 0xcd, 0x7e, 0x81, 0xfc, 0x62, 0x3b, 0x2b, 0x4b, 0x7d, 0x13, 0x33};
+  byte_t tx_id0[IOTA_TRANSACTION_ID_BYTES] = {126, 127, 95,  249, 151, 44,  243, 150, 40,  39, 46,
+                                              190, 54,  49,  73,  171, 165, 88,  139, 221, 25, 199,
+                                              90,  172, 252, 142, 91,  179, 113, 2,   177, 58};
+
+  // create transaction payload
+  uint16_t network_id = 2;
+  transaction_payload_t* tx = tx_payload_new(network_id);
 
-  transaction_payload_t* tx = tx_payload_new();
-  TEST_ASSERT_NOT_NULL(tx);
+  // add input with tx_id0
+  TEST_ASSERT(tx_essence_add_input(tx->essence, 0, tx_id0, 1) == 0);
 
-  TEST_ASSERT(tx_payload_add_input_with_key(tx, tx_id0, 0, seed_keypair.pub, seed_keypair.priv) == 0);
-  TEST_ASSERT(tx_payload_add_output(tx, OUTPUT_SINGLE_OUTPUT, addr0, 1000) == 0);
-  TEST_ASSERT(tx_payload_add_output(tx, OUTPUT_SINGLE_OUTPUT, addr1, 2779530283276761) == 0);
+  // add basic output one
+  output_basic_t* basic_output_one = create_output_basic_one();
+  TEST_ASSERT(tx_essence_add_output(tx->essence, OUTPUT_BASIC, basic_output_one) == 0);
+
+  // add basic output two
+  output_basic_t* basic_output_two = create_output_basic_two();
+  TEST_ASSERT(tx_essence_add_output(tx->essence, OUTPUT_BASIC, basic_output_two) == 0);
 
-  // put tx payload into message
-  msg->payload_type = 0;
+  // create message
+  uint8_t protocol_ver = 2;
+  core_message_t* msg = core_message_new(protocol_ver);
+  TEST_ASSERT_NOT_NULL(msg);
   msg->payload = tx;
+  msg->payload_type = CORE_MESSAGE_PAYLOAD_TRANSACTION;
+
+  core_message_print(msg, 0);
 
-  TEST_ASSERT(core_message_sign_transaction(msg) == 0);
+  // calculate transaction essence hash
+  byte_t essence_hash[CRYPTO_BLAKE2B_HASH_BYTES] = {};
+  TEST_ASSERT(core_message_essence_hash_calc(msg, essence_hash, sizeof(essence_hash)) == 0);
 
-  tx_payload_print(tx);
+  // check if essence is matching
+  TEST_ASSERT_EQUAL_MEMORY(expected_essence_hash, essence_hash, CRYPTO_BLAKE2B_HASH_BYTES);
```

Comment:
except for essence hash, it should validate Unlock Block via `signing_transaction_sign` as well.

Reply:
Done.